### ISSUE TITLE
Make app updates automatic instead of downloading from the site

### ISSUE THREAD

**Issue by MohanSha** [will-stone/browserosaurus#211]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I have to manually download and install the app updates from the site

**Describe the solution you'd like**
Download the update automatically and prompt to install reducing manual work to update

**Describe alternatives you've considered**
None

**Additional context**
The updates were better handled before the redesign, it would automatically download the update the icon would turn blue in the bar indicating there is an update. So when I click update the app exits and updates itself and starts

**Comment by will-stone** [comment]
v9 used the [Electron Autoupdater](https://www.electronjs.org/docs/api/auto-updater) which will always download the update, and install it when restarting the app. This isn't great, as you'll have people who would like to stay on a certain version.

The ideal solution would be that the Autoupdater had a method that only checks, a method that only downloads, and so on. Therefore, I would suggest either opening an issue with Electron, or I am happy to look at a PR if you would like to merge both my solution and the Autoupdater.

Remember, behind every "frustrated" download you make are feature and bug fixes I have added to the app `<cough>`for free`</cough>` 😁

**Comment by brettgoss** [comment]
Is an opt-out setting for 'Automatically download updates' an option to implement? I feel like that's how most electron apps using autoUpdater handle it. Though that would require a preferences pane...

If that's something we want to do, then Insomnia has a good example of how they handle it: https://github.com/Kong/insomnia/blob/develop/packages/insomnia-app/app/main/updates.js

**Comment by will-stone** [comment]
Interesting. It seems like wrapping the Autoupdater would benefit the whole Electron community; anyone fancy making a package? Or seeing if anyone has already started a project for this? Sounds like something Sindre Sorhus would do, not sure if he's still in the Electron game any more.

**Comment by will-stone** [comment]
Looks like this is what we need: https://www.electron.build/auto-update I’m not sure how compatible with Electron Forge / Packager it is though.

**Comment by xlogix** [comment]
Yes, this is very frustrating. Please give it a priority... Thank you for this app!

**Comment by will-stone** [comment]
Okay... I give it a low priority; the current solution works and is very easy to maintain. If this is important to you, please consider contributing a PR.

**Comment by xlogix** [comment]
I would love to. Just very new to this... But, I'll try

**Comment by will-stone** [comment]
Awesome. My free time is about to come to an end (have been waiting to start a new job) so your help is appreciated. Get something into a draft PR and I can help out.

**Comment by vildhjarta8** [comment]
+1, also want this feature back.

@will-stone, please :)

**Comment by mbaierl** [comment]
Ha, just wanted to ask the same... for an auto update feature. Great that it is already being worked on. Thanks!

**Comment by will-stone** [comment]
I have some free time today so I'll look at adding this now. Please remember that if you are unable to help by submitting code, please consider [sponsorhip](https://github.com/sponsors/will-stone): the lowest tier is just $1 a month.

**Comment by will-stone** [comment]
Okay, findings so far: the `electron-updater` does not work well with Electron Forge. I'm getting an error:

```
Error: Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (85:14)
    at Object.module.exports.pp$4.raise ...
```

Which isn't particularly useful. I've spent the last hour trying to find a fix but cannot find anything that helps.

The reason I wanted to use this package is that it breaks up the update process so users can restart the app without a forced update, i.e. stay on certain versions. I think I'll go back to the standard autoUpdater and use the Insomnia method, linked to by @brettgoss, where a setting can be used to prevent the loading of the autoUpdater on app load. I'm going to add the autoUpdater now and worry about where this setting will live later. If I make a major change before then (to v11) then I'll just do what I did with v9 and release a version with the autoUpdater removed (I'm highly unlikely to do this for some time!).
